java/flight/flight-core/src/main/java/org/apache/arrow/flight/client/ClientCookieMiddleware.java

```diff
@@ -74,25 +75,32 @@ public void onHeadersReceived(CallHeaders incomingHeaders) {
     // to signal that the client should stop using the cookie immediately.
     final Consumer<String> handleSetCookieHeader = (headerValue) -> {
       final List<HttpCookie> parsedCookies = HttpCookie.parse(headerValue);
-      parsedCookies.forEach(parsedCookie -> cookies.put(parsedCookie.getName(), parsedCookie));
+      parsedCookies.forEach(parsedCookie -> factory.cookies.put(parsedCookie.getName(), parsedCookie));
     };
-    incomingHeaders.getAll(SET_COOKIE_HEADER).forEach(handleSetCookieHeader);
-    incomingHeaders.getAll(SET_COOKIE2_HEADER).forEach(handleSetCookieHeader);
+    final Iterable<String> setCookieHeaders = incomingHeaders.getAll(SET_COOKIE_HEADER);
+    if (setCookieHeaders != null) {
+      for (String cookieHeader : setCookieHeaders) {
```

Comment:
setCookieHeaders.forEach(handleSetCookieHeader)

Reply:
Done, that looks cleaner.